roles/translator/src/status.rs

```diff
@@ -0,0 +1,46 @@
+use crate::error::Error;
+
+#[derive(Debug)]
+pub enum State<'a> {
+    Shutdown(Error<'a>),
+    Healthy(String),
+}
+
+#[derive(Debug)]
+pub struct Status<'a> {
+    pub state: State<'a>,
+}
+
+/// # Description
+/// This macro handles errors inserting error handling logic for a given `Result<T, crate::error::Error<'a>>`
+/// it is used by passing in a `Sender<crate::status::Status>` as the first parameter
+/// and a `Result<T, crate::error::Error<'a>>` as the second parameter.
+/// NOTE: can only be used within async functions since status needs to be send over async channel
+/// # Example
+/// ```
+/// let (tx_status: Sender<Status>, rx_status) = async_channel::unbounded();
+/// let variable = handle_result!(
+///     tx_status,
+///     type_.try_into()
+/// );
+/// ```
+#[macro_export]
+macro_rules! handle_result {
+    ($sender:expr, $res:expr) => {
+        match $res {
+            Ok(val) => val,
+            Err(e) => {
+                tracing::error!("Error: {:?}", e);
+                // send status
+                $sender
+                    .send($crate::status::Status {
+                        state: $crate::status::State::Shutdown(e.into()),
```

Comment:
We should be able to:
1. handle the error in the context where it has been generated 
2. handle the error at "main" level and maybe shutdown the thread/context that generate the error
3. handle the error at "main" level and exit

To me it seem that right now we have only (3) is that right? Maybe (1) is not necessary but I think that (2) is very necessary.

Reply:
For the examples that we have now they all do shutdown - but my thinking was eventually the main thread could potentially recover a task based on the State/Status/Error sent. Right now we haven't figured out what tasks can be restarted or which ones should be restarted so for now we are just shutting everything down "gracefully". You are correct though in saying that we should be able to do all 3 eventually; that's the goal.